### PR TITLE
[1629] Fix other course length

### PR DESCRIPTION
**Pull request by dankmitchell**
### Context
Setting "other" course length

### Changes proposed in this pull request
Pass across the "other" length into `course_length` when the user has selected it

### Guidance to review
Try setting "other" for course length

### CODE REVIEW

**Comment by dankmitchell** [comment]
[[M] Updating course length doesnt work](https://trello.com/c/eEF5TILv/1629-m-updating-course-length-doesnt-work)

File: app/controllers/courses_controller.rb [hunk 1]
```diff
@@ -11,6 +11,11 @@ def index; end
   def show; end
 
   def update
+    # Course length should be saved as `course_length` so if "other" is selected then pass that text value into `course_length`
+    if params[:course][:course_length_other_length].present? && params[:course][:course_length] == 'Other'
```
Comment (defong) [inline app/controllers/courses_controller.rb]: can we use dig

```ruby
params.dig(:course, :course_length_other_length).present? && params.dig(:course, :course_length) == 'Other'
```

File: spec/features/courses/fees_spec.rb [hunk 2]
```diff
@@ -64,6 +64,28 @@
     expect(current_path).to eq description_provider_course_path('AO', course_1.course_code)
   end
 
+  context 'with course_length_other selected' do
+    let(:course_1) do
+      jsonapi(
+        :course,
+        :with_fees,
+        course_length: '6months',
```
Comment (defong) [inline spec/features/courses/fees_spec.rb]:
```suggestion
        course_length: '6 months',
```

closer to real life example

**Comment by fofr** [comment]
When you first select "Other", it should be empty, it shouldn't show our internal representation of the value:

other-field-not-blank

**Pull request merged**